File: src/Servers/PresenceConnectionManager/test/UniSpyServer.Servers.PresenceConnectionManager.Test/Buddy/BuddyRequestTest.cs
using Xunit;
using UniSpyServer.Servers.PresenceConnectionManager.Entity.Structure.Request.Buddy;

namespace UniSpyServer.Servers.PresenceConnectionManager.Test.Buddy
{
    public class BuddyRequestTest
    {
        [Fact]
        public void AddBuddy()
        {
            var request = new AddBuddyRequest(BuddyRequests.AddBuddy);
            request.Parse();
            Assert.Equal((uint)0, request.FriendProfileID);
            Assert.Equal("test", request.Reason);
        }

        [Fact]
        public void DelBuddy()
        {
            var request = new DelBuddyRequest(BuddyRequests.DelBuddy);
            request.Parse();
            Assert.Equal((uint)0, request.DeleteProfileID);
        }

        [Fact]
        public void InviteTo()
        {
            var request = new InviteToRequest(BuddyRequests.InviteTo);
            request.Parse();
            Assert.Equal((uint)0, request.ProductID);
            Assert.Equal((uint)0, request.ProfileID);
        }

        // TODO:
        /*[Fact]
        public void StatusInfo()
        {
            var request = new StatusInfoRequest(BuddyRequests.StatusInfo);
            request.Parse();
            Assert.Equal((uint)0, request.ProfileID);
            Assert.Equal((uint)0, request.NamespaceID);
        }*/

        [Fact]
        public void Status()
        {
            var request = new StatusRequest(BuddyRequests.Status);
            request.Parse();
            Assert.Equal("test", request.Status.StatusString);
            Assert.Equal("test", request.Status.LocationString);
        }
    }
}

File: src/servers/presence_connection_manager/tests/buddy_request_test.py
import unittest

from servers.presence_connection_manager.contracts.requests.buddy import (
    AddBuddyRequest,
    DelBuddyRequest,
    InviteToRequest,
    StatusRequest,
)


class BuddyRequestTest(unittest.TestCase):
    ADD_BUDDY = "\\addbuddy\\\\sesskey\\0\\newprofileid\\0\\reason\\test\\final\\"
    DEL_BUDDY = "\\delbuddy\\\\sesskey\\0\\delprofileid\\0\\final\\"
    INVITE_TO = "\\inviteto\\\\sesskey\\0\\productid\\0\\profileid\\0\\final\\"
    STATUS = "\\status\\0\\statstring\\test\\locstring\\test\\final\\"

    def test_add_buddy(self) -> None:
        request = AddBuddyRequest(BuddyRequestTest.ADD_BUDDY)
        request.parse()
        self.assertEqual(0, request.friend_profile_id)
        self.assertEqual("test", request.reason)

    def test_del_buddy(self) -> None:
        request = DelBuddyRequest(BuddyRequestTest.DEL_BUDDY)
        request.parse()
        self.assertEqual(0, request.friend_profile_id)

    def test_invite_to(self) -> None:
        request = InviteToRequest(BuddyRequestTest.INVITE_TO)
        request.parse()
        self.assertEqual(0, request.product_id)
        self.assertEqual(0, request.profile_id)

    def test_status_test(self) -> None:
        request = StatusRequest(BuddyRequestTest.STATUS)
        request.parse()
        self.assertEqual("test", request.status.status_string)
        self.assertEqual("test", request.status.location_string)


if __name__ == "__main__":
    unittest.main()
